test/generators/typescript/preset/__snapshots__/ExamplePreset.spec.ts.snap

```diff
@@ -0,0 +1,69 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Example function generation should render example function for model 1`] = `
+"export class Test {
+  private _stringProp: string;
+  private _numberProp?: number;
+  private _objectProp?: NestedTest;
+  private _additionalProperties?: Map<String, object | string | number | Array<unknown> | boolean | null | number>;
+  private _sTestPatternProperties?: Map<String, string>;
+
+  constructor(input: {
+    stringProp: string,
+    numberProp?: number,
+    objectProp?: NestedTest,
+  }) {
+    this._stringProp = input.stringProp;
+    this._numberProp = input.numberProp;
+    this._objectProp = input.objectProp;
+  }
+
+  get stringProp(): string { return this._stringProp; }
+  set stringProp(stringProp: string) { this._stringProp = stringProp; }
+
+  get numberProp(): number | undefined { return this._numberProp; }
+  set numberProp(numberProp: number | undefined) { this._numberProp = numberProp; }
+
+  get objectProp(): NestedTest | undefined { return this._objectProp; }
+  set objectProp(objectProp: NestedTest | undefined) { this._objectProp = objectProp; }
+
+  get additionalProperties(): Map<String, object | string | number | Array<unknown> | boolean | null | number> | undefined { return this._additionalProperties; }
+  set additionalProperties(additionalProperties: Map<String, object | string | number | Array<unknown> | boolean | null | number> | undefined) { this._additionalProperties = additionalProperties; }
+
+  get sTestPatternProperties(): Map<String, string> | undefined { return this._sTestPatternProperties; }
+  set sTestPatternProperties(sTestPatternProperties: Map<String, string> | undefined) { this._sTestPatternProperties = sTestPatternProperties; }
+
+  public static example(): Test {
+    const instance = new Test({} as any);
+    instance.stringProp = \\"string\\";
+    instance.numberProp = 0;
+    instance.objectProp = NestedTest.example();
+    return instance;
+  }
```

Comment:
Did you think about using the `openapi-sampler` or something like this? I understand that you wanna add examples directly to the props, but maybe `openapi-sampler` + Object.assign(...) will make the implementation easier, what do you think? I know that problem will be with nested objects, like in your example for `NestedTest`.

Reply:
We could, but... remember each generator can define their naming behavior, meaning a property name, is not the same as the JSON property name.

One could do, which I currently am in the ts-nats-template use `openapi-sampler` and the un/marshal functionality, but that creates a direct dependency and unnecessary complexity in my opinion.

Reply:
ok :) Someone can also write his own preset to achieve similar output based on `openapi-sampler`.

Reply:
Yes, exactly 👍